sdk/storage/storage-blob/src/utils/utils.common.ts

```diff
@@ -186,19 +186,17 @@ export function extractConnectionStringParts(connectionString: string): Connecti
       url: blobEndpoint,
       accountName,
       accountKey,
-      proxyUri
+      proxyUri,
     };
   } else {
     // SAS connection string
 
     let accountSas = getValueInConnString(connectionString, "SharedAccessSignature");
-    let accountName = getAccountNameFromUrl(blobEndpoint);
+    let accountName = getValueInConnString(connectionString, "AccountName");
```

Comment:
This isn't right. This branch is to deal with "SAS connection string" https://github.com/Azure/azure-sdk-for-js/pull/4632 which looks like this "BlobEndpoint=https://myaccount.blob.core.windows.net/;QueueEndpoint=https://myaccount.queue.core.windows.net/;FileEndpoint=https://myaccount.file.core.windows.net/;TableEndpoint=https://myaccount.table.core.windows.net/;SharedAccessSignature=sasString".
I wonder how can we get the account name in this case with custom domain. Or we don't need it in this case?
And why isn't our test case breaking for this?

Reply:
Thanks for the suggestion!